cmd/kube-controller-manager/app/certificates.go

```diff
@@ -36,6 +40,46 @@ func startCSRSigningController(ctx ControllerContext) (bool, error) {
 	if ctx.Options.ClusterSigningCertFile == "" || ctx.Options.ClusterSigningKeyFile == "" {
 		return false, nil
 	}
+
+	// Deprecation warning for old defaults.
+	//
+	// * If the signing cert and key are the default paths but the files
+	// exist, warn that the paths need to be specified explicitly in a
+	// later release and the defaults will be removed. We don't expect this
+	// to be the case.
+	//
+	// * If the signing cert and key are default paths but the files don't exist,
+	// bail out of startController without logging.
+	filesExist := true
+	usesDefaults := true
+
+	_, err := os.Stat(ctx.Options.ClusterSigningCertFile)
+	filesExist = filesExist && !os.IsNotExist(err)
+
+	usesDefaults = usesDefaults && (ctx.Options.ClusterSigningCertFile == options.DefaultClusterSigningCertFile)
+
+	_, err = os.Stat(ctx.Options.ClusterSigningKeyFile)
+	filesExist = filesExist && !os.IsNotExist(err)
+
+	usesDefaults = usesDefaults && (ctx.Options.ClusterSigningKeyFile == options.DefaultClusterSigningKeyFile)
+
+	switch {
+	case filesExist && usesDefaults:
```

Comment:
Do you want this warning to appear if they only are using one of the two defaults? Because that wouldn't be the case as written.

Reply:
Yes I do.